x-pack/plugins/uptime/public/components/settings/add_connector_flyout.tsx

```diff
@@ -30,16 +30,15 @@ export const AddConnectorFlyout = ({ focusInput }: Props) => {
 
   const dispatch = useDispatch();
 
-  useEffect(() => {
-    dispatch(getConnectorsAction.get());
-    focusInput();
-  }, [addFlyoutVisible, dispatch, focusInput]);
-
   const ConnectorAddFlyout = useMemo(
     () =>
       getAddConnectorFlyout({
         consumer: 'uptime',
-        onClose: () => setAddFlyoutVisibility(false),
+        onClose: () => {
+          dispatch(getConnectorsAction.get());
+          setAddFlyoutVisibility(false);
+          focusInput();
+        },
       }),
     // eslint-disable-next-line react-hooks/exhaustive-deps
```

Comment:
Not a part of this PR, but why are we disabling exhaustive deps? It's generally not recommended by the hooks team.

Reply:
i also don't have much context, this particular piece was added by alerting team.

As far i remember, it was rendering component twice , which was causing some flickering behaviours.

Reply:
I played around with adding the dependencies, and I did not see a flicker. I console.log'd inside the useMemo to make sure it was only ever firing once on component mount, and played around with touching all the buttons inside the flyout. Perhaps recent refactors have made this no longer necessary?

Reply:
in any case, we can do that as follow up. I don't want to increase the scope of this PR.